src/runtime/testing/handle-for-test.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {UnifiedStore} from '../storageNG/unified-store.js';
+import {Arc} from '../arc.js';
+import {SingletonHandle, CollectionHandle} from '../storageNG/handle.js';
+import {ActiveStore} from '../storageNG/store.js';
+import {Flags} from '../flags.js';
+import {StorageProxy} from '../storageNG/storage-proxy.js';
+import {handleFor, Singleton, Storable, Collection} from '../handle.js';
+import {Referenceable, CRDTCollectionTypeRecord} from '../crdt/crdt-collection.js';
+import {CRDTTypeRecord} from '../crdt/crdt.js';
+import {CRDTSingletonTypeRecord} from '../crdt/crdt-singleton.js';
+
+/**
+ * Creates a singleton handle for a store for testing purposes. Returns an
+ * appropriate OldHandle/HandleNG type depending on the storage migration flag.
+ */
+// TODO: Can we correctly type the result here?
+// tslint:disable-next-line: no-any
+export async function singletonHandleForTest(arc: Arc, store: UnifiedStore): Promise<SingletonHandle<any>> {
```

Comment:
SingletonHandle is a new handle. Is it better to return the Handle class (which is new|old)?

Reply:
Callers want to be able to call methods specific to the handle subclasses (e.g. get/set for singletons, add/remove/toList for collections).

Additionally, those methods for old vs. new classes are slightly different. The old Collection class has a store method, but the new CollectionHandle class calls it add. And for Singleton vs. SingletonHandle, they have slightly different types in their function signature.

My plan was to migrate everything to use the new Handle API, and create shim functions (e.g. collectionHandleWrapper) so that you can use the old classes with the new API.